tt_metal/impl/kernels/runtime_args_data.hpp

```diff
@@ -16,19 +21,60 @@ struct RuntimeArgsData {
     std::uint32_t * rt_args_data;
     std::size_t rt_args_count;
 
-    std::uint32_t & operator[](std::size_t index);
+    inline bool in_bounds(std::size_t index) const noexcept {
+        if(index >= rt_args_count) [[unlikely]] {
```

Comment:
Question on why we are trying to avoid having TT_FATAL in headers?
Seems like we'll end up duplicating assertion code formatting if anything in our header files needs to assert, and may lead to inconsistent formatting down the line?

Reply:
TT_FATAL and TT_ASSERT `should be` internal logging implementation details.
Both depend on `libfmt`, which again, should be a private/internal dependency.

`runtime_args_data.hpp` happens to be a public header file, because it gives the definition of `RuntimeArgsData` to `host_api.h`.

If we keep our internal logging macros in public header files... 
We leak our internal logging implementation details, and force a dependency `fmt` on customer.

What if customer code uses a different version of fmt?

Reply:
I see. Is this the only public header that was using these assertions? Just concerned about consistency/duplication down the line

Reply:
Probably not, but I had to start somewhere. :-)